datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -380,6 +376,12 @@ def save_node(
     node.current_version = node_revision.version
     node_revision.extra_validation()
 
+    if node_revision.type == NodeType.METRIC:
```

Comment:
Your PR summary does not mention this being specific for METRIC nodes. I understand this may be expensive to maintain for all node columns, but now when I think of measures coming (possibly to the picture) the doing this for Metrics only may stop making sense.

Reply:
@agorajek Good point, I can add lineage for all other nodes as well. It's not actually that expensive assuming that we only do it once at node creation/update time. If it is too time consuming, I was also thinking about turning it into something that's a background task that gets kicked off async when the node is created or updated so that it's non-blocking.

Reply:
@agorajek Good point, I can add lineage for all other nodes as well. It's not actually that expensive assuming that we only do it once at node creation/update time. If it is too time consuming, I was also thinking about turning it into something that's a background task that gets kicked off async when the node is created or updated so that it's non-blocking.

Reply:
@shangyian I think having a lineage on every column of every node is the ultimate goal of this. So if you think this is doable I'd add it from the start and worry about the performance later.

Reply:
@agorajek Done! I excluded source and cube nodes though, as those aren't quite the same pattern as the other node types that have queries.